test/kaniko_task_test.go

```diff
@@ -0,0 +1,185 @@
+	return &v1alpha1.PipelineResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kanikoResourceName,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PipelineResourceSpec{
+			Type: v1alpha1.PipelineResourceTypeGit,
+			Params: []v1alpha1.Param{
+				v1alpha1.Param{
+					Name:  "Url",
+					Value: "https://github.com/pivotal-nader-ziada/gohelloworld",
+				},
+			},
+		},
+	}
+}
+
+func getTask(namespace string, t *testing.T) *v1alpha1.Task {
+	dockerRepo := os.Getenv("DOCKER_REPO_OVERRIDE")
+	if dockerRepo == "" {
+		t.Fatalf("DOCKER_REPO_OVERRIDE env variable is required")
+	}
+
+	return &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      kanikoTaskName,
+		},
+		Spec: v1alpha1.TaskSpec{
+			Inputs: &v1alpha1.Inputs{
+				Resources: []v1alpha1.TaskResource{
+					v1alpha1.TaskResource{
+						Name: kanikoResourceName,
+						Type: v1alpha1.PipelineResourceTypeGit,
+					},
+				},
+			},
+			BuildSpec: &buildv1alpha1.BuildSpec{
+				Timeout: &metav1.Duration{Duration: 2 * time.Minute},
+				Steps: []corev1.Container{{
+					Name:  "kaniko",
+					Image: "gcr.io/kaniko-project/executor",
+					Args: []string{"--dockerfile=/workspace/Dockerfile",
+						fmt.Sprintf("--destination=%s/kanikotasktest", dockerRepo),
+						"--no-push"},
+				}},
+			},
+		},
+	}
+}
+
+func getTaskRun(namespace string) *v1alpha1.TaskRun {
+	return &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      kanikoTaskRunName,
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name: kanikoTaskName,
+			},
+			Trigger: v1alpha1.TaskTrigger{
+				TriggerRef: v1alpha1.TaskTriggerRef{
+					Type: v1alpha1.TaskTriggerTypeManual,
+				},
+			},
+			Inputs: v1alpha1.TaskRunInputs{
+				Resources: []v1alpha1.PipelineResourceVersion{
+					v1alpha1.PipelineResourceVersion{
+						ResourceRef: v1alpha1.PipelineResourceRef{
+							Name: kanikoResourceName,
+						},
+						Version: "master",
+					},
+				},
+			},
+		},
+	}
+}
+
+// TestTaskRun is an integration test that will verify a TaskRun using kaniko
+func TestKanikoTaskRun(t *testing.T) {
+	logger := logging.GetContextLogger(t.Name())
+	c, namespace := setup(t, logger)
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(logger, c.KubeClient, namespace) }, logger)
+	defer tearDown(logger, c.KubeClient, namespace)
+
+	if _, err := c.PipelineResourceClient.Create(getGitResource(namespace)); err != nil {
+		t.Fatalf("Failed to create Pipeline Resource `%s`: %s", kanikoResourceName, err)
+	}
+
+	// Create task
+	if _, err := c.TaskClient.Create(getTask(namespace, t)); err != nil {
+		t.Fatalf("Failed to create Task `%s`: %s", kanikoTaskName, err)
+	}
+
+	// Create TaskRun
+	if _, err := c.TaskRunClient.Create(getTaskRun(namespace)); err != nil {
+		t.Fatalf("Failed to create TaskRun `%s`: %s", kanikoTaskRunName, err)
+	}
+
+	// Verify status of TaskRun (wait for it)
```

Comment:
instead of using comments here (and whenever it seems reasonable), can we add log output that says something similar? e.g. `logger.Infof("Waiting for TaskRun %s to complete", tr.Name)`

For long running tests it can really help to have output like this so we don't think it's just hanging

Reply:
ok